docs/source/api_reference/utils.rst

```diff
@@ -708,6 +708,22 @@ The utilities here are used to modify the attributes of the classes. Used by dif
 .. autoclass:: deepchem.utils.attribute_utils.del_attr
   :members:
 
+Polymer Weighted Directed Graph Data Utilities
+-----------------------------------------
+
+These classes and functions are required to handle converstion of string data to graph data 
+and validation of the same. 
+
+Polymer Weighted Directed Graph String Validator
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This class validates the input string representation for graphical conversion of polymer data.
+It splits the input strings, validates their order and values. Upon recieving error it can return
+meaningful error message to the user indicating the type of error that hinders the conversion process.
```

Comment:
Add citations, examples. This is a new input format so it needs a lot of references/explanation